File: Bakery/Core/Model/Bun/Bun.cs
using System;

namespace Bakery.Core.Model.Bun
{
    public class Bun : BunBase
    {
        public Bun()
        {
        }

        public Bun(Bun bun)
            : this(bun.BakeTime, bun.SellUntil, bun.TargetSaleTime, bun.OriginalPrice)
        {
        }

        public Bun(DateTime bakeTime, DateTime sellUntil, DateTime targetSaleTime, decimal price)
            : base(bakeTime, sellUntil, targetSaleTime, price)
        {
        }
    }
}

File: Bakery/ClientApp/src/app/models/bun.ts
import { BunType } from "./bun-type";

export class Bun {
    constructor(
        public id: number,
        public bunType: BunType|number,
        public originalPrice: number,
        public currentPrice: number,
        public sellUntil: Date,
        public targetSaleTime: Date,
        public bakeTime: Date,
        public nextPriceChangeTime: Date,
        public nextPrice: number)
    {
    }
}